homeassistant/components/velbus.py

```diff
@@ -0,0 +1,71 @@
+"""
+Support for Velbus platform.
+
+For more details about this platform, please refer to the documentation at XXX
+"""
+import asyncio
+import logging
+import velbus
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.core import callback
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+REQUIREMENTS = ['python-velbus==2.0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'velbus'
+
+VELBUS_MESSAGE = 'velbus.message'
+
+CONFIG_MODULE = vol.Schema({
+    vol.Required('module'): cv.positive_int,
+    vol.Required('channel'): cv.positive_int,
+    vol.Required('name'): cv.string,
+    vol.Optional('is_pushbutton'): cv.boolean
+})
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required('serial_port'): cv.string,
+    vol.Optional('lights'): vol.All(cv.ensure_list, [CONFIG_MODULE]),
+    vol.Optional('switches'): vol.All(cv.ensure_list, [CONFIG_MODULE])
+})
+
+
+@asyncio.coroutine
+def async_setup(hass, config):  # noqa: D401
+    """Setup the Velbus platform."""
+    conf = config.get(DOMAIN)[0]
+    device = conf.get('serial_port')
+
+    connection = velbus.VelbusUSBConnection(device)
+    controller = velbus.Controller(connection)
```

Comment:
For the users it would be helpful if the exceptions are caught in case of connection or setup failures. The setup should return with a log entry otherwise the users will end up with a non-functional component.